functions/src/modules/SE.functions.ts

```diff
@@ -88,3 +93,276 @@ export const processNewAllotment = functions.database
+                    topics: row['Topics'],
+                    gurudevaTimings: row['Gurudeva Timings'],
+                    otherSpeakers: row['Other Speakers'],
+                    kirtan: row['Kirtan'],
+                    abruptLecture: row['Abrupt Lecture'],
+                    suggestedTitle: row['Suggested Title'],
+                    languages: row['Languages'],
+                    soundQualityRating: row['Sound Quality'],
+                    soundIssues: row['Sound Issues'],
+                    comments: row['Comments'],
+                    submissionTimestamp: row['Timestamp'],
+                    submissionSerial: row['Submission Serial'],
+                },
+                importTimestamp: admin.database.ServerValue.TIMESTAMP,
+                processingResolution: row['Resolution'],
+                _import
+            };
+
+
+            
+            // Same file name --> add newly read chunk (row) to the `currentFile` object
+            if (currentFile.file_name === AudioFileName) 
+                currentFile.chunks.push(chunkToStore);
+            
+            // NEW FILE --> Save old file AND add the newly read chunk to `currentFile` object            
+            else {
+                if (currentFile.file_name != null) {
+
+                    // if file object exists, add the chunks to it
+                    if (chunks[currentFile.file_name]) 
+                        chunks[currentFile.file_name].push(currentFile.chunks)
+
+                    // if this is the first occurence of the file, set its value to the list of chunks 
+                    else    
+                        chunks[currentFile.file_name] = currentFile.chunks;
+                }
+
+                currentFile.file_name = AudioFileName;
+                currentFile.chunks = [chunkToStore];
+            }                    
+        }
+        
+
+        ////////////////////////////////
+        // Processing Sheet Audio Files
+        ////////////////////////////////
+
+        Object.keys(chunks).forEach(async fileName => {
+            let currentFilechunks = chunks[fileName];
+
+            // Sorting by `beginning` time
+            currentFilechunks = await lodash.orderBy(currentFilechunks, ['beginning'], ['asc']);
+            // console.log(currentFilechunks);
+
+            let chunksToImport = [];
+            ////////////////////////////////////////////
+            // [CHECK #5] Chunks of a single audio file 
+            // should not overlap each other in timing.
+            ////////////////////////////////////////////
+            let lastEndingTime = null;
+            currentFilechunks.forEach((chunk, i) => {
+
+
+                // `Beginning` value should be less than `Ending`.
+                if (chunk.beginning > chunk.ending) {
+                    console.log(`Audio file name: ${fileName} -- Error: "Beginning" value isn't less than "Ending".`);
+                    summary.overlappingChunks.push({
```

Comment:
This is not “Chunks of a single audio file should not overlap each other in timing” case. This is “Beginning value should be less than Ending” case.

Overlapping means that two chunks are overlapping each other. Example:

- 00:00-15:00
- 10:00-20:00

Reply:
sorry, forgot to add them to a different category